src/test/java/hudson/plugins/ec2/EC2RetentionStrategyTest.java

```diff
@@ -238,6 +243,28 @@ public void testTerminateOfflineComputerIfNotConnecting() throws Exception {
         assertThat(logging.getMessages(), hasItem(containsString("offline but not connecting, will check if it should be terminated because of the idle time configured")));
     }
 
+
+    /**
+     * Do not terminate na instance if a computer just launched, and the
+     * retention strategy timeout has not expired yet. The idle timeout
+     * is correctly accounted for nodes that have been stopped and started
+     * again (i.e termination policy is stop, rather than terminate).
+     */
+    @Test
+    public void testDoNotTerminateInstancesJustBooted() throws Exception {
```

Comment:
This test is insufficient, it passes with or without this patch

Reply:
yeah, let me see what I can do about it

Reply:
we should be good now, thanks!